### PR TITLE
Show what game(s) a sound is from

### LINKED ISSUE #1220: Sound Browser - Show what game a sound is from
In the Sound Browser, it is impossible to know what game a specific sound is from. If I'm building something I'll share later, I want to avoid sounds that don't come with Garry's Mod itself. If it's the Heavy shouting something about a sandwich I obviously know that's from Team Fortress 2, but other times it's not that easy to tell.

### PR DESCRIPTION
**Pull request by TomyLobo**
Would be nice if someone could test this, preferably with a lot of mountable games :)

* Does a lot of cleanup
* Potentially also fixes some kind of right-click functionality in the sound browser tree. (2nd to last commit)
* Fixes #1220 (in the last commit)
* Bases on #1320

### CODE REVIEW

**Comment by TomyLobo** [comment]
If this is too slow with a lot of games, maybe it can be loaded dynamically when opening the "Games" node somehow...

**Comment by TomyLobo** [comment]
The failing checks are due to glua

**Comment by TomyLobo** [comment]
Now the remaining 26 warnings are unused variables/arguments

**Comment by Nebual** [comment]
Don't put both refactoring of code style and bug fixes in one PR >:~O

**Comment by AbigailBuccaneer** [comment]
I'd say it's fine to do the same in one PR, just not in one commit.

File: lua/wire/client/sound_browser.lua [hunk 1]
```diff
@@ -87,6 +87,18 @@ local function GetInfoTable(strfile)
 	return T, not tabproperty
 end
 
+local function GetGamesForSound(path)
+	local games = {}
+
+	for _,game in pairs(engine.GetGames()) do
+		if not game.mounted then return end
+		if not file.Exists(path, game.folder) then return end
```
Comment (bigdogmat) [inline lua/wire/client/sound_browser.lua]: Shouldn't this just be a check of whether to insert into games, not return?

Comment (TomyLobo) [inline lua/wire/client/sound_browser.lua]: oops, you're right. I had previously only returned one value :)
I'll fix it right away

Comment (TomyLobo) [inline lua/wire/client/sound_browser.lua]: Fixed

**Comment by TomyLobo** [comment]
Ok i fixed what @bigdogmat found and rebased the whole thing onto latest master.
Btw, I used "continue", what's our policy on that?

**Comment by Divran** [comment]
Continue is the only good in the C syntax things that garry added. I don't see anything wrong with using it, but as you know you could get around it simply by encasing the contents of your loop with an if statement to get the same behavior.

**Comment by bigdogmat** [comment]
> I don't see anything wrong with using it

I agree, although that makes it non standard Lua syntax which goes against [coding style](https://github.com/wiremod/wire/wiki/Coding-style#other-stuff).

**Comment by TomyLobo** [comment]
There are cases where the resulting code isn't simple at all, but yeah, it's non-standard so we can't use it.

**Comment by thegrb93** [comment]
I don't think this is possible to do anymore. file.Exist works differently now

**Comment by thegrb93** [comment]
If you want the style fixes merged though, we can do that.

**Comment by thegrb93** [comment]
Closes: #1220

**Pull request closed without merging**